mlpp_lib/normalizers.py

```diff
@@ -0,0 +1,519 @@
+        dataset: xr.Dataset,
+        variables: Optional[list] = None,
+        dims: Optional[list] = None,
+    ):
+        if variables is None:
+            variables = list(dataset.data_vars)
+        if not all(var in dataset.data_vars for var in variables):
+            raise KeyError(
+                f"There are variables not in dataset: {[var for var in variables if var not in dataset.data_vars]}"
+            )
+
+        self.minimum = dataset[variables].min(dims).compute().copy()
+        self.maximum = dataset[variables].max(dims).compute().copy()
+        self.fillvalue = self.fillvalue
+
+    def transform(
+        self, *datasets: xr.Dataset, variables: Optional[list] = None
+    ) -> tuple[xr.Dataset, ...]:
+        if self.minimum is None:
+            raise ValueError("MinMaxScaler wasn't fit to data")
+
+        def f(ds: xr.Dataset, variables: Optional[list] = None) -> xr.Dataset:
+
+            ds_copy = ds.copy()
+            if variables is None:
+                variables = ds_copy.data_vars
+            for var in variables:
+                assert var in self.minimum.data_vars, f"{var} not in MinMaxScaler"
+
+                scaled_var = (
+                    (ds_copy[var] - self.minimum[var])
+                    / (self.maximum[var] - self.minimum[var])
+                ).astype("float32")
+
+                if self.fillvalue is not None:
+                    scaled_var = scaled_var.fillna(self.fillvalue)
+
+                ds_copy[var] = scaled_var
+            return ds_copy
+
+        return tuple(f(ds, variables) for ds in datasets)
+
+    def inverse_transform(
+        self, *datasets: xr.Dataset, variables: Optional[list] = None
+    ) -> tuple[xr.Dataset, ...]:
+        if self.minimum is None:
+            raise ValueError("MinMaxScaler wasn't fit to data")
+
+        def f(ds: xr.Dataset, variables: Optional[list] = None) -> xr.Dataset:
+            if variables is None:
+                variables = ds.data_vars
+
+            ds = xr.where(ds > self.fillvalue, ds, np.nan)
+            for var in variables:
+                assert var in self.minimum.data_vars, f"{var} not in MinMaxScaler"
+
+                unscaled_var = (
+                    ds[var] * (self.maximum[var] - self.minimum[var])
+                    + self.minimum[var]
+                ).astype("float32")
+                ds[var] = unscaled_var
+
+            return ds
+
+        return tuple(f(ds, variables) for ds in datasets)
+
+    @classmethod
+    def from_dict(cls, in_dict: dict) -> Self:
+        minimum = xr.Dataset.from_dict(in_dict["minimum"])
+        maximum = xr.Dataset.from_dict(in_dict["maximum"])
+        return cls(minimum, maximum)
+
+    def to_dict(self):
+        out_dict = {
+            "minimum": self.minimum.to_dict(),
+            "maximum": self.maximum.to_dict(),
+            "fillvalue": self.fillvalue,
+        }
+        return out_dict
+
+
+@dataclass
+class MaxAbsScaler(DataTransformation):
```

Comment:
shouldn't this rather be called "AbsMaxScaler"?

Reply:
Not sure. To me since we compute the maximum of the absolute values, it was clearer as MaxAbs. What do you think ?